clients/drcachesim/tracer/output.cpp

```diff
@@ -1335,33 +1347,25 @@ exit_thread_io(void *drcontext)
 #endif
 
     if (is_in_tracing_mode(tracing_mode.load(std::memory_order_acquire)) ||
-        (has_tracing_windows() && !op_split_windows.get_value()) ||
+        (has_tracing_windows() &&
+         (!op_split_windows.get_value() ||
+          (get_local_window(data) < tracing_window.load(std::memory_order_acquire) &&
```

Comment:
So when writing windows to different files, now we emit an exit event even if there was nothing in that window? (or maybe I'm reading this wrong)

In general, it'd help to extract individual conditions into separate bools, to avoid too much nesting inside the if-condition. Of course except cases where the complete condition must not be evaluated before.

Reply:
No, the other way around: if non-split we need to append a thread exit (when split, the exit is inserted at the end of each separate file).

I added a top-level comment and individual comments: I think it is much clearer now.